pkg/config/types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2017 Kinvolk GmbH
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+const (
+	RuntimeDocker = "docker"
+	RuntimeRkt    = "rkt"
+	RuntimeCrio   = "crio"
+)
+
+// ClusterConfiguration holds the state of a cluster
+// with all the information needed to
+// run kube-spawn
+//
+type ClusterConfiguration struct {
+	KubeSpawnDir string `toml:"-" mapstructure:"dir"`
+
+	Name              string `toml:"cluster-name" mapstructure:"cluster-name"`
+	ContainerRuntime  string `toml:"container-runtime" mapstructure:"container-runtime"`
+	KubernetesVersion string `toml:"kubernetes-version" mapstructure:"kubernetes-version"`
+	Image             string `toml:"image" mapstructure:"image"`
+	Nodes             int    `toml:"nodes" mapstructure:"nodes"`
+
+	// DevCluster indicates if we should run
+	// from a local kubernetes build
+	DevCluster bool `toml:"dev" mapstructure:"dev"`
+
+	RuntimeConfiguration RuntimeConfiguration `toml:"runtime-config,omitempty" mapstructure:"runtime-config"`
+
+	// Files to be copied from host to overlay of all machines
+	// map: dst = src
+	// not saved into kspawn.toml
+	Copymap map[string]string `toml:"-"`
```

Comment:
Is there any reason to do it like `dst = src`? Because to me it looks intuitive to have the other way around, `src = dst`.

Reply:
Yeah I thought about both ways... I ended up going with this because it feels like variable assignment to me. 
But I am open to changing it if that makes no sense.

Reply:
@robertgzr Ok, if you say so, I don't have strong opinion.